app/controllers/ethnic_churches_controller.rb

```diff
@@ -13,13 +13,7 @@ def new
   end
 
   def create
-    @ethnic_church = EthnicChurch.new(ethnic_church_params)
-    languages = params[:language][:name]
-    @ethnic_church.languages = languages
-    @ethnic_church.country = Country.find_or_initialize_by(name: country_params[:name])
-    @ethnic_church.religious_background = ReligiousBackground.find_or_initialize_by(persuasion: religious_background_params[:persuasion])
-    @ethnic_church.build_address(address_params)
-    @ethnic_church.build_note(note_params)
+    @ethnic_church = EthnicChurchBuilder.instantiate(params)
```

Comment:
Instead of passing in all the params that rails gives u try narrowing it down to what you need.

For what i can see u can do something like
``` ruby
EthnicChurchBuilder.instantiate(
languages: languages,
country_name: country_name,
address: address,
note_content: note_content
```
where 
 ``` 
languages
country_name
address 
note_content
``` 
are private methods within your controller that do what they already do in your EthnicChurchBuilder class

for instance 
``` ruby
private # in controller
def languages
 @params[:language][:name].reject(&:blank?)
end
```

Reply:
I like the way this makes the params more explicit. I'm going to run with this!

Reply:
@ericmathison indeed! i like strong params because it tells me what i need to pass in. This can also help with refactoring when u start to see that u require more params in there. Thats where dependency injection comes to play :)